src/dotnet/commands/dotnet-publish/PublishCommandParser.cs

```diff
@@ -49,6 +49,23 @@ internal static class PublishCommandParser
                             string value = o.Arguments.Any() ? o.Arguments.Single() : "true";
                             return $"-property:SelfContained={value}";
                         })),
+                Create.Option(
+                    "--no-self-contained",
```

Comment:
I don't think we should copy the bad design of having a boolean option take an argument, like how `--self-contained` unfortunately operates (which is very un-POSIX like).

I think `--no-self-contained` should not take an option and always imply `--self-contained false` (i.e. forwarded as `-property:SelfContained=false`).

See the `--no-build` option as an example of this.

Reply:
Agree --no-self-contained, no argument. 

```
--no-self-contained false 
```

makes my head hurt